src/sentry/consumers/__init__.py

```diff
@@ -163,109 +164,122 @@ def ingest_monitors_options() -> list[click.Option]:
 ]
 
 # consumer name -> consumer definition
-# XXX: default_topic is needed to lookup the schema even if the actual topic name has been
-# overridden. This is because the current topic override mechanism means the default topic name
-# is no longer available anywhere in code. We should probably fix this later so we don't need both
-#  "topic" and "default_topic" here though.
+# XXX: topic_override will be deprecated here in favor of KAFKA_TOPIC_OVERRIDE once all overrides in
+# prod have been migrated
 KAFKA_CONSUMERS: Mapping[str, ConsumerDefinition] = {
     "ingest-profiles": {
-        "topic": settings.KAFKA_PROFILES,
+        "topic": Topic.PROFILES,
+        "topic_override": settings.KAFKA_PROFILES,
```

Comment:
Isn't this supposed to be replaced by the new `KAFKA_TOPIC_OVERRIDES` ?
I assume this is still referencing the old setting for backward compatibility during the rollout ?

Reply:
my interpretation is that we will gradually remove `topic_override` lines per-consumer, and right now this allows us to merge this PR before KAFKA_TOPIC_OVERRIDES is populated

it's confusing to me though that `KAFKA_TOPIC_OVERRIDES` is the new thing while `topic_override` is deprecated

Reply:
Yeah, this is temporary thing that is going away, and aligns to how it works today (it was just named `topic` before). Though because of the consumers defined in getsentry there will need to be multiple even more temporary steps before this :(